plugins/out_stackdriver/stackdriver_http_request.c

```diff
@@ -0,0 +1,453 @@
+        msgpack_pack_true(mp_pck);
+    }
+    else {
+        msgpack_pack_false(mp_pck);
+    }
+}
+
+/* latency should be in the format: 
+ *      whitespace (opt.) + integer + point & decimal (opt.)
+ *      + whitespace (opt.) + "s" + whitespace (opt.) 
+ */
+static void validate_latency(msgpack_object_str latency_in_payload, 
+                             struct http_request_field *http_request) {
+    flb_sds_t pattern = flb_sds_create("^\\s*\\d\\d*(.\\d\\d*)?\\s*s\\s*$");
+    char extract_latency[latency_in_payload.size];
+    struct flb_regex *regex;
+
+    int status = 0;
+    int i = 0, j = 0;
+
+    regex = flb_regex_create(pattern);
+    status = flb_regex_match(regex, latency_in_payload.ptr, latency_in_payload.size);
+    flb_regex_destroy(regex);
+    flb_sds_destroy(pattern);
+
+    if (status == 1) {
+        for (; i < latency_in_payload.size; ++ i) {
+            if (latency_in_payload.ptr[i] == '.' || latency_in_payload.ptr[i] == 's' 
+                || isdigit(latency_in_payload.ptr[i])) {
+                extract_latency[j] = latency_in_payload.ptr[i];
+                ++ j;
+            }
+        }
+        http_request->latency = flb_sds_copy(http_request->latency, extract_latency, j);
+    }
+}
+                                                                                        
+/* Return true if httpRequest extracted */
+int extract_http_request(struct http_request_field *http_request, 
+                         msgpack_object *obj, int *extra_subfields)
+{
+    http_request_status op_status = NO_HTTPREQUEST;
+    msgpack_object_kv *p;
+    msgpack_object_kv *pend;
+    msgpack_object_kv *tmp_p;
+    msgpack_object_kv *tmp_pend;
+
+    if (obj->via.map.size == 0) {
+        return FLB_FALSE;
+    }
+                                                                                        
+    p = obj->via.map.ptr;
+    pend = obj->via.map.ptr + obj->via.map.size;
+                                                                                        
+    for (; p < pend && op_status == NO_HTTPREQUEST; ++p) {
+        if (p->val.type != MSGPACK_OBJECT_MAP
+            || p->key.type != MSGPACK_OBJECT_STR
+            || sizeof(HTTPREQUEST_FIELD_IN_JSON) - 1 != p->key.via.str.size
+            || strncmp(HTTPREQUEST_FIELD_IN_JSON, 
+                       p->key.via.str.ptr, p->key.via.str.size) != 0) {
+            
+            continue;
+        }
+
+        op_status = HTTPREQUEST_EXISTS;
+        msgpack_object sub_field = p->val;
+        
+        tmp_p = sub_field.via.map.ptr;
+        tmp_pend = sub_field.via.map.ptr + sub_field.via.map.size;
+                                                                                        
+        /* Validate the subfields of httpRequest */
+        for (; tmp_p < tmp_pend; ++tmp_p) {
+            if (tmp_p->key.type != MSGPACK_OBJECT_STR) {
+                continue;
+            }
+
+            if (sizeof("requestMethod") - 1 == tmp_p->key.via.str.size
```

Comment:
can we group by the type of the value, and then verify the key of the field? This way reduces lines, and it is consistent with implementation of other functions as well.

Also checking key alphabetical within the same value type as the implementation in other parts.

Reply:
Done